pkg/api/common/versions.go

```diff
@@ -198,37 +198,99 @@ var AllKubernetesSupportedVersions = map[string]bool{
 	"1.19.0-beta.0":  true,
 }
 
+// AllKubernetesSupportedVersionsOnAzureStack is a whitelist map of all supported Kubernetes version strings on Azure Stack
+// The bool value indicates if creating new clusters with this version is allowed
+var AllKubernetesSupportedVersionsOnAzureStack = map[string]bool{
+	"1.14.7":  false,
+	"1.14.8":  false, // disabled because of https://github.com/Azure/aks-engine/issues/2312
+	"1.15.4":  false,
+	"1.15.5":  false, // disabled because of https://github.com/Azure/aks-engine/issues/2312
+	"1.15.10": false,
+	"1.15.11": true,
+	"1.15.12": true,
+	"1.16.8":  false,
+	"1.16.9":  true,
+	"1.16.10": false,
+	"1.17.4":  false,
+	"1.17.5":  true,
+	"1.17.6":  false,
+}
+
+// AllKubernetesWindowsSupportedVersionsOnAzureStack maintain a set of available k8s Windows versions in aks-engine on Azure Stack
+var AllKubernetesWindowsSupportedVersionsOnAzureStack = map[string]bool{
+	"1.15.10": false,
+	"1.15.11": true,
+	"1.15.12": true,
+	"1.16.8":  false,
+	"1.16.9":  true,
+	"1.17.4":  false,
+	"1.17.5":  true,
+}
+
 // GetDefaultKubernetesVersion returns the default Kubernetes version, that is the latest patch of the default release
-func GetDefaultKubernetesVersion(hasWindows bool) string {
-	defaultRelease := KubernetesDefaultRelease
-	if hasWindows {
-		defaultRelease = KubernetesDefaultReleaseWindows
+func GetDefaultKubernetesVersion(hasWindows bool, onAzureStack bool) string {
+	var defaultRelease string
+	if onAzureStack {
+		if hasWindows {
+			defaultRelease = KubernetesDefaultReleaseWindowsAzureStack
+		} else {
+			defaultRelease = KubernetesDefaultReleaseAzureStack
+		}
+	} else {
+		if hasWindows {
+			defaultRelease = KubernetesDefaultReleaseWindows
+		} else {
+			defaultRelease = KubernetesDefaultRelease
+		}
 	}
```

Comment:
Effectively the same, but I prefer this better
```
defaultRelease := KubernetesDefaultRelease
if hasWindows {
	defaultRelease = KubernetesDefaultReleaseWindows
}
if isAzureStack && hasWindows {
	defaultRelease = KubernetesDefaultReleaseWindowsAzureStack
}
if isAzureStack && !hasWindows {
	defaultRelease = KubernetesDefaultReleaseAzureStack
}
```
I think it would be more consistent with the code base (and probably slightly easier to read) if the "Azure" code is executed regardless and then we override if target is Azure Stack

Reply:
addressed

Reply:
addressed